### PR TITLE
Add AlmaLinux to the familiy of Red Hat-alike operating systems

### PR DESCRIPTION
**Pull request by cschug**
##### SUMMARY

This change adds support for [AlmaLinux](https://almalinux.org/) and adds it as a member to the `RedHat` family of operating systems.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/facts/system/distribution.py

### CODE REVIEW

**Comment by webknjaz** [comment]
@cschug why is the commit credited to another account?

**Comment by cschug** [comment]
> @cschug why is the commit credited to another account?

Yeah, that is something which also makes me curious. I own both accounts (cschug & fundsaccess-dco) but they have the same GPG key deployed. Apparently Github makes some funny normalization here.

**Comment by webknjaz** [comment]
I believe they match the `Author` and `Committer` fields with GH accounts that have respective emails verified, it shouldn't rely on GPG AFAIK.

**Comment by cschug** [comment]
Ah, that might make sense. Will try to fix that in the future. Does this cause any issues for this specific PR?

**Comment by webknjaz** [comment]
Not really but I thought you may want to do `git commit --amend --author='Another Name <with@email.there>'` and `git push --force-with-lease` to update it.

**Review by samdoran — CHANGES_REQUESTED** [review]
To make sure we get this working correctly, can you please generate test fixtures using `hacking/tests/gen_distribution_version_testcase.py` and add unit tests to `test/units/module_utils/facts/system/distribution/test_distribution_version.py`? Depending on the contents of the release file, we may need to add more methods to `distributon.py`.

This also needs a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

File: changelogs/fragments/support_almalinux.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- Add AlmaLinux to the familiy of Red Hat-alike operating systems
```
Comment (samdoran) [inline changelogs/fragments/support_almalinux.yml]: I would consider this a bugfix, which will also allow the change to be backported.
```suggestion
bugfixes:
  - add AlmaLinux to fact gathering (https://github.com/ansible/ansible/pull/73458)
```

**Review by samdoran — CHANGES_REQUESTED** [review]

**Comment by cschug** [comment]
@samdoran Thanks for your support and your guidance on this matter so far, I very much appreciate it. I cannot see how I can resolve the failed CI jobs. I cannot relate those failures to this code change, to me they rather look like issues of the CI infrastructure (authentication issues and 50 minute timeouts exceeded). Do you have any hint how to progress any further? Thanks in advance!

**Comment by samdoran** [comment]
This just looks like new changes were merged into the branch rather than rebasing the changes in this branch on a newer upstream branch.

**Comment by relrod** [comment]
@cschug Do you plan on cleaning/finishing this PR? If not, I can take it finish it up.

**Comment by cschug** [comment]
@relrod, no I won't continue on that.

I simply lack the time to spend hours and hours on this very simple basically one-line change. I thought it might be a good idea to get this included in the upstream so that also others can benefit from it. I don't need this change not really because I package all my software as RPMs on my own (I've got 20 years of experience in RPM packaging, so it was a no time job to get this change included into my RPM).

But I have little experience with Python and especially the test framework stuff. I tried neverthless to make somehow sense of all that stuff and commit this here little by little but found that this CI pipeline super-frustrating. It is just painful slow and with one single exception broke *every single time* because of some infrastructure problems (timeouts, request rate limits exceeded, authentication and certificate problems). I'm just curious if it's just me having bad karma or if it affects others as well.

Somehow those broken build jobs resolved automagically by just waiting yet another day. The whole process is simply not transparent (to me). Also the default output of just "Bash exited with code '1'." or whatever is not helpful at all, unless you dive into this sluggish Azure  Pipeline web interface thingy which requires so many resources that my browser and my little notebook's CPU cannot handle it. It's basically one click -> 3 minutes of waiting. Just to figure out several clicks and cups of coffee later that there apparently was an infrastructure problem *again*, not a code related breakage. With no obvious way to retrigger that job.

The latest CI breakage it also did not resolve automagically as before (which BTW I also do not expect to happen, I rather found this magic super scary) and was just sitting in broken state for days. Therefore I somehow had the idea to rebase/squash the stuff and to retrigger it again for yet another attempt. By doing so I made a mistake, but on the other hand realized how much time I already invested in the trivial thing. For sure it is a problem on my side because apparently it works for others. But I have no idea how to resolve the situation and gave up on something, what I don't actually need. The cost/benefit ratio did not work out on my side. Sorry. That's why I closed this PR and abandoned the idea to get this merged.

If you want to finish it up, feel free ... and good luck.

**Comment by samdoran** [comment]
I'll fix up the commit history. We had some CI issues yesterday that should now be resolved.

**Comment by samdoran** [comment]
Merged into `devel`. Thanks for your contributor @cschug!

**Comment by samdoran** [comment]
I also opened backports for 2.10 and 2.9.

**Pull request closed without merging**